dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -60,12 +63,27 @@
     Entry parseWorkflowContent(Entry entry, String filepath, String content, Set<SourceFile> sourceFiles);
 
     /**
-     * Confirms whether the content of a descriptor contains a valid workflow
-     *
-     * @param content the content of a descriptor
-     * @return true iff the workflow looks like a valid workflow
+     * Validates a workflow set for the workflow described by with primaryDescriptorFilePath
+     * @param sourcefiles Set of sourcefiles
+     * @param primaryDescriptorFilePath Primary descriptor path
+     * @return Is a valid workflow set, error message
      */
-    boolean isValidWorkflow(String content);
+    ImmutablePair validateWorkflowSet(Set<SourceFile> sourcefiles, String primaryDescriptorFilePath);
```

Comment:
Since this is in an interface (and gets implemented and called a few times) as opposed to a private method
This is probably worth the cost of creating real inner classes to represent the return object (with proper named getter and setters) to make the code elsewhere more readable